Save m.local_notification_settings.<device-id> event in account_data (PSG-873)

mnaturel opened the pull request:
<!-- Please read [CONTRIBUTING.md](https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md) before submitting your pull request -->
 
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

<!-- Describe shortly what has been changed -->
When background sync is used, on session start and each time we switch on background sync, account data event "org.matrix.msc3890.local_notification_settings.<device-id>" is updated.
When switching from background sync to regular push notifications method, any existing account data event is updated with empty content (API to delete an account data event is not available yet).

Also in this PR: renaming some notifications related use cases for consistency.

## Motivation and context

<!-- Provide link to the corresponding issue if applicable or explain the context -->
Closes #7596 

## Screenshots / GIFs

<!-- Only if UI have been changed
You can use a table like this to show screenshots comparison.
Uncomment this markdown table below and edit the last line `|||`:
|copy screenshot of before here|copy screenshot of after here|
-->

<!--
|Before|After|
|-|-|
|||
 -->

## Tests

<!-- Explain how you tested your development -->

Tests should be performed with both the GPlay and FDroid variant.
- Install Ntfy app
- Open the Network plugin in Flipper
- Check the network call is correct after signin
- Check the network call is correct when toggling the notifications for the current session in the settings
- (Only FDroid) Check the network call is correct when switching from background sync mode to push notification method changing the related setting (Notifications -> Notifications method)

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): Android 11

## Checklist

<!-- Depending on the Pull Request content, it can be acceptable if some of the following checkboxes stay unchecked. -->

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

mnaturel commented on vector/src/main/java/im/vector/app/features/settings/devices/v2/notification/UpdateNotificationSettingsAccountDataUseCase.kt:
```diff
@@ -0,0 +1,60 @@
+) {
+
+    suspend fun execute(session: Session) {
+        if (unifiedPushHelper.isBackgroundSync()) {
```
I am not sure this is the correct method to use to check if background sync is enabled. To be tested further to ensure it is working as expected.

mnaturel replied:
I have changed the implementation so that we can rely on this method.

mnaturel commented:
In order to have correct behavior, I need to first fix some bugs on [the notification method selection process](https://github.com/vector-im/element-android/issues/7653).

ganfra reviewed (approved):
LGTM, thanks for all the tests.

mnaturel commented:
@ganfra Could you take a look again please? I had to change several things which were not working correctly. I have also rebased my work on [this PR in which I refactored some logic for unifiedPush to fix several issues](https://github.com/vector-im/element-android/pull/7675).
